src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -283,21 +283,22 @@ public virtual void WriteResponseHeaders(OutputFormatterWriteContext context)
         /// <returns>The mediaType with the given encoding.</returns>
         protected string GetMediaTypeWithCharset(string mediaType, Encoding encoding)
         {
-            var mediaTypeEncoding = MediaTypeEncoding.GetEncoding(mediaType);
-            if (mediaTypeEncoding != null &&
-                mediaTypeEncoding.WebName.Equals(encoding.WebName, StringComparison.OrdinalIgnoreCase))
+            if (mediaType == null)
             {
-                return mediaType;
+                throw new ArgumentNullException(nameof(mediaType));
             }
-            else
+
+            if (encoding == null)
             {
-                if (encoding.WebName.Equals(Encoding.UTF8.WebName, StringComparison.OrdinalIgnoreCase))
-                {
-                    return OutputMediaTypeCache[mediaType];
-                }
+                throw new ArgumentNullException(nameof(encoding));
+            }
 
-                return MediaTypeEncoding.ReplaceEncoding(mediaType, encoding);
+            if (encoding.WebName.Equals(Encoding.UTF8.WebName, StringComparison.OrdinalIgnoreCase))
+            {
+                return OutputMediaTypeCache[mediaType];
```

Comment:
That doesn't really mean anything. This code would call that code. I'm fine with leaving it here, but let's not pretend that it enables :sparkler: **Magic.** :tophat: